source-code/website/src/pages/marketplace/@id/index.page.tsx

```diff
@@ -0,0 +1,193 @@
+import { Meta, Title } from "@solidjs/meta"
+import { Layout } from "#src/pages/Layout.jsx"
+import { Markdown, parseMarkdown } from "#src/services/markdown/index.js"
+import { For, Show, createEffect, createSignal } from "solid-js"
+import type { ProcessedTableOfContents } from "./index.page.server.jsx"
+import { GetHelp } from "#src/components/GetHelp.jsx"
+import { marketplaceItems } from "@inlang/marketplace"
+import type { MarketplaceItem } from "@inlang/marketplace"
+import { Button } from "#src/pages/index/components/Button.jsx"
+import { Chip } from "#src/components/Chip.jsx"
+import MaterialSymbolsArrowOutward from "~icons/material-symbols/arrow-outward"
+import { SelectRepo } from "../SelectRepo.jsx"
+
+/**
+ * The page props are undefined if an error occurred during parsing of the markdown.
+ */
+export type PageProps = {
+	processedTableOfContents: ProcessedTableOfContents
+	markdown: Awaited<ReturnType<typeof parseMarkdown>>
+	urlPathname: string
+}
+
+export function Page(props: PageProps) {
+	const [item, setItem] = createSignal<MarketplaceItem | undefined>(undefined)
+	const [readmore, setReadmore] = createSignal<boolean>(false)
+
+	/* Automatically sets the fitting item e.g. when changing bundle */
+	createEffect(() => {
+		setItem(
+			marketplaceItems.find(
+				(item) =>
+					item.meta.displayName.en?.toLowerCase().replaceAll(" ", "-") ===
+					props.markdown?.frontmatter?.title?.toLowerCase().replaceAll(" ", "-"),
+			),
+		)
+	})
```

Comment:
@floriandwt use derived state. You don't need an effect for derived state. If you find yourself calling `set*` in an effect, you are likely using the effect wrong. 

Please change the code to used derived state

```ts
const item = () => marketplaceItems.find((item) => ...)
```